### PR TITLE
Stripping trailing commas and spaces from `vec!` elements

### PR DESCRIPTION
**Pull request by bekh6ex**
Fixes issues https://github.com/rust-lang-nursery/rustfmt/issues/878 and https://github.com/rust-lang-nursery/rustfmt/issues/1034

### CODE REVIEW

**Comment by bekh6ex** [comment]
As far as I understood discussion in the issues mentioned above, we should not remove trailing commas because it can change the behaviour.

How about we will reformat only `vec!` macros? 
It won't change code's behaviour and will be pretty useful, as soon as it is one of the most popular ones.

File: src/macros.rs [hunk 1]
```diff
@@ -114,7 +114,11 @@ pub fn rewrite_macro(mac: &ast::Mac,
             parser.bump();
 
             if parser.token == Token::Eof {
-                return None;
+                if macro_name == "vec!" {
```
Comment (nrc) [inline src/macros.rs]: Could you add a comment here to explain the special cases

Comment (bekh6ex) [inline src/macros.rs]: Added

**Comment by nrc** [comment]
Thank you!

**Pull request merged**